eZ/Publish/Core/REST/Server/Input/Parser/Criterion/LogicalAnd.php

```diff
@@ -30,12 +30,20 @@ class LogicalAnd extends CriterionParser
      */
     public function parse(array $data, ParsingDispatcher $parsingDispatcher)
     {
+        if (array_key_exists('LogicalAnd', $data) && is_array($data['LogicalAnd'])) {
+            $data['AND'] = $data['LogicalAnd'];
+        }
+
         if (!array_key_exists('AND', $data) && !is_array($data['AND'])) {
             throw new Exceptions\Parser('Invalid <AND> format');
         }
 
         $criteria = array();
         foreach ($data['AND'] as $criterionName => $criterionData) {
+            if (is_array($criterionData)) {
+                $criterionName = key($criterionData);
+                $criterionData = current($criterionData);
+            }
```

Comment:
So it turns out that the syntax we've initially planned: 
```json
{
    "AND": [
        {
            "Field": {  }
        },
        {
            "Field": {  }
        }
    ]
}
```
was not really supported, right? I'm all for this new feature as it seems it won't affect existing payloads. But maybe whole syntax for logical criterions is supposed to be different. 

@bdunogier do you have some input on that? It could be of course that we didn't design operators to use multiple criterions of the same name, but I just want to make sure.